crates/router/src/core/routing.rs

```diff
@@ -909,26 +909,43 @@ pub async fn retrieve_default_fallback_algorithm_for_profile(
 }
 
 #[cfg(feature = "v1")]
-
 pub async fn retrieve_default_routing_config(
     state: SessionState,
+    profile_id: Option<common_utils::id_type::ProfileId>,
     merchant_account: domain::MerchantAccount,
     transaction_type: &enums::TransactionType,
 ) -> RouterResponse<Vec<routing_types::RoutableConnectorChoice>> {
     metrics::ROUTING_RETRIEVE_DEFAULT_CONFIG.add(&metrics::CONTEXT, 1, &[]);
     let db = state.store.as_ref();
-
-    helpers::get_merchant_default_config(
-        db,
-        merchant_account.get_id().get_string_repr(),
-        transaction_type,
-    )
-    .await
-    .map(|conn_choice| {
-        metrics::ROUTING_RETRIEVE_DEFAULT_CONFIG_SUCCESS_RESPONSE.add(&metrics::CONTEXT, 1, &[]);
-        service_api::ApplicationResponse::Json(conn_choice)
-    })
+    if let Some(profile_id) = profile_id {
+        helpers::get_merchant_default_config(db, profile_id.get_string_repr(), transaction_type)
+            .await
+            .map(|conn_choice| {
+                metrics::ROUTING_RETRIEVE_DEFAULT_CONFIG_SUCCESS_RESPONSE.add(
+                    &metrics::CONTEXT,
+                    1,
+                    &[],
```

Comment:
Can we handle the functions in the if and else block in a common manner .

Reply:
Okay so let's consider this situation when the function `get_merchant_default_config`, returns error.
So the metrics(`ROUTING_RETRIEVE_DEFAULT_CONFIG_SUCCESS_RESPONSE`) will be pushed even when we are returning an error, which shouldn't be the case.

So considering this situation the metrics is being pushed only when we have a success confirmation.

Reply:
No I am talking about the function `get_merchant_default_config` move it out of the if else block